ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,359 @@
+    "/simulation/service/shower_distribution",
+]
+optional_nodes = [
+    "/simulation/service/shower_distribution",
+    "/simulation/event/telescope/images",
+    "/simulation/event/telescope/parameters",
+]
+allowlist_simu = [
+    "/simulation/event/subarray/shower",
+    "/simulation/event/telescope/parameters",
+    "/simulation/event/telescope/images",
+    "/simulation/service/shower_distribution",
+]
+allowlist_service = [
+    "/dl1/service/image_statistics",
+    "/dl1/service/image_statistics.__table_column_meta__",
+]
+allowlist_tels = [
+    "/dl1/monitoring/telescope/pointing",
+    "/dl1/event/telescope/parameters",
+    "/dl1/event/telescope/images",
+    "/simulation/event/telescope/parameters",
+    "/simulation/event/telescope/images",
+]
+blocklist_images = ["/simulation/event/telescope/images", "/dl1/event/telescope/images"]
+blocklist_parameters = [
+    "/simulation/event/telescope/parameters",
+    "/dl1/event/telescope/parameters",
+]
+
+
+class MergeTool(Tool):
+    name = "ctapipe-merge"
+    description = "Merges DL1-files from the stage1-process tool"
+    examples = """
+    To merge DL1-files created by the stage1-process-tool from current directory:
+
+    > ctapipe-merge file1.h5 file2.h5 file3.h5 --ouput=/path/output_file.h5 --progress
+
+    If you want a specific file pattern as input files use --pattern='pattern.*.dl1.h5', e.g.:
+
+    > ctapipe-merge --output=/path/output_file.h5 --progress --pattern='pattern.*.dl1.h5'
+
+    If neither any input files nor any pattern is given, all files from the current directory
+    with the default pattern '*.h5' are taken. For merging all files from a specific
+    directory with a given pattern, use:
+
+    > ctapipe-merge --input-dir=/input/dir/ --output=/path/output_file.h5 --progress
+    --pattern='pattern.*.dl1.h5'
+    """
+    input_dir = traits.Path(
+        help="input dl1-directory", exists=None, directory_ok=True, file_ok=False
+    ).tag(config=True)
+    input_files = List(default_value=[], help="input dl1-files").tag(config=True)
+    output_path = traits.Path(help="merged-DL1 output filename").tag(config=True)
+    skip_images = traits.Bool(
+        help="skip /dl1/event/telescope/images and "
+        "/simulation/event/telescope/images data in output",
+        default_value=False,
+    ).tag(config=True)
+    skip_simu_images = traits.Bool(
+        help="skip /simulation/event/telescope/images " "data in output",
+        default_value=False,
+    ).tag(config=True)
+    skip_parameters = traits.Bool(
+        help="skip /dl1/event/telescope/parameters and "
+        "/simulation/event/telescope/parameters "
+        "data in output",
+        default_value=False,
+    ).tag(config=True)
+    overwrite = traits.Bool(help="overwrite output file if it exists").tag(config=True)
+    progress_bar = traits.Bool(help="show progress bar during " "processing").tag(
+        config=True
+    )
+    file_pattern = traits.Unicode(
```

Comment:
Is a glob pattern enough here or do we need a regex?

Reply:
Can't think for now of a good use case for regexp where glob pattern doesn't work or is too intricate, do you?